lib/vast/Dialect/HighLevel/HighLevelOps.cpp

```diff
@@ -41,8 +43,146 @@ namespace vast::hl
 {
     using FoldResult = mlir::OpFoldResult;
 
+
+    //===----------------------------------------------------------------------===//
+    // ArithBinOps
+    //===----------------------------------------------------------------------===//
+
+    FoldResult checked_int_arithmetic(
+        mlir_type type, auto adaptor, auto &&op
+    ) {
+        if (auto lhs = mlir::dyn_cast_or_null< core::IntegerAttr >(adaptor.getLhs())) {
```

Comment:
I would do
```
auto rhs = ...
auto lhs = ...
if (!rhs || !lhs)
  return {};
return op(rhs, lhs);
```
I think it is in a sense cleaner as that `rhs/lhs` nesting is not really "dependent" on each step.